frameworks/helloworld/src/main/java/com/mesosphere/sdk/helloworld/scheduler/Main.java

```diff
@@ -233,4 +238,47 @@ private static ServiceSpec createSampleServiceSpec(SchedulerConfig schedulerConf
                         .build())
                 .build();
     }
+
+    private static void checkAndMigrate(
+            FrameworkConfig frameworkConfig,
+            SchedulerConfig schedulerConfig,
+            Persister persister
+    ) {
+        /**
+         * Notes:
+         *
+         * 1. We do not migrate in Dynamic Service Mode because if we do so, there is a "gap" between the scheduler
+         * registering with mesos and it receiving inbound requests with ymls. During this "gap" if it receives any
+         * offers, it does not know what to do with them if we support migration. Hence we support migration only in
+         * static service mode.
+         * 2. Note that if the {{user}} field in the yml is not specified in the mono mode, we used to default to `root`
+         * In the multi mode however, the framework config takes precedence if a user is not mentioned. Caution must
+         * be taken to ensure the service user is same (mention it explicitly in both mono and multi for the sake of
+         * consistency) during the migration.
+         * 3. The migration MUST happen before the {@link SchedulerBuilder#build()} is called (i.e., before the
+         * scheduler talks to {@link Persister}.
+         */
+        if (!schedulerConfig.isMonoToMultiMigrationDisabled()) {
+            LOGGER.info("Migration is enabled. Analyzing now...");
+            SchemaVersionStore schemaVersionStore = new SchemaVersionStore(persister);
+            int curVer = schemaVersionStore.getOrSetVersion(SchemaVersionStore.getSupportedSchemaVersionMultiService());
+            if (curVer == SchemaVersionStore.getSupportedSchemaVersionSingleService()) {
+                try {
+                    LOGGER.warn("Found old schema in ZK Storage that can be migrated to a new schema");
+                    PersisterUtils.backUpFrameworkZKData(persister);
+                    PersisterUtils.migrateMonoToMultiZKData(persister, frameworkConfig);
+                    schemaVersionStore.store(SchemaVersionStore.getSupportedSchemaVersionMultiService());
+                    LOGGER.info("Successfully migrated from old schema to new schema!!");
+                } catch (PersisterException e) {
+                    LOGGER.error("Unable to migrate ZK data : ", e.getMessage(), e);
```

Comment:
When an exception is being passed, we would need to do our own string formatting via `String.format()`. The call being used here is:
```
     * @param format the format string
     * @param arg1   the first argument
     * @param arg2   the second argument
     */
    public void error(String format, Object arg1, Object arg2);
```
However, the one you would want here is:
```
     * @param msg the message accompanying the exception
     * @param t   the exception (throwable) to log
     */
    public void error(String msg, Throwable t);
```

... but in this case, this could just be of this form: `LOGGER.error("Unable to migrate ZK data", e);`